modules/core/shared/src/main/scala/io/circe/JsonObject.scala

```diff
@@ -275,17 +275,21 @@ object JsonObject {
    * Note that the order of the fields is arbitrary.
    */
   final def fromMap(map: Map[String, Json]): JsonObject =
-    if (map.isEmpty) {
-      empty
-    } else {
-      fromMapAndVector(map, map.keys.toVector)
-    }
+    fromMapAndVector(map, map.keys.toVector)
 
   private[circe] final def fromMapAndVector(map: Map[String, Json], keys: Vector[String]): JsonObject =
-    new MapAndVectorJsonObject(map, keys)
+    (map.size: @switch) match {
```

Comment:
Same critique on size above, use sizeCompare or sizeIs instead so a full traversal isn't used when it isn't needed.

Reply:
thanks, changed to `sizeIs`